lesson4/js/main.js

```diff
@@ -1,18 +1,27 @@
 "use strict";
 
 const ul = document.getElementById("js-ul");
-const li = document.createElement("li");
+const fragment = document.createDocumentFragment();
 
-function makeLiElm(elm) {
+const attributes = [
+  { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+  { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" },
+];
+
+attributes.forEach((attribute) => {
   const li = document.createElement("li");
-  li.innerHTML = elm;
-  ul.appendChild(li);
-}
+  const a = document.createElement("a");
+  const img = document.createElement("img");
+
+  a.href = attribute.to;
+  img.src = attribute.img;
+  img.alt = attribute.alt;
+
+  a.textContent = attribute.text;
+  a.insertBefore(img, a.firstChild);
 
-makeLiElm("<a href='a1.html'><img src='/img/bookmark.png'>a1</a>");
-makeLiElm("<a href='a2.html'><img src='/img/message.png'>a2</a>");
+  li.appendChild(a);
+  fragment.appendChild(li);
```

Comment:
IMO
It is recommended to use [Method chaining](https://dev.to/isiakaabd/method-chaining-in-javascript-154a).

```suggestion
  fragment.appendChild(li).appendChild(a);
```

Reply:
Thank you!
I had no idea about method chains💦.